### PR TITLE
FLP-394 Added logic to handle creation of employer initiated price ch…

### PR DESCRIPTION
**Pull request by tom-gough**
…ange request in inner api.

### CODE REVIEW

File: src/Command/AddPriceHistory/CreateApprenticeshipPriceChangeCommandHandler.cs [hunk 1]
```diff
@@ -23,7 +23,19 @@ public CreateApprenticeshipPriceChangeCommandHandler(IApprenticeshipRepository a
             CancellationToken cancellationToken = default)
         {
             var apprenticeship = await _apprenticeshipRepository.Get(command.ApprenticeshipKey);
-            apprenticeship.AddPriceHistory(command.TrainingPrice, command.AssessmentPrice, command.TotalPrice, command.EffectiveFromDate, DateTime.Now, PriceChangeRequestStatus.Created, command.UserId, command.Reason);
+            if (string.Equals(command.Requester, PriceChangeRequester.Provider.ToString(), StringComparison.CurrentCultureIgnoreCase))
```
Comment (MikeV-TalentConsulting) [inline src/Command/AddPriceHistory/CreateApprenticeshipPriceChangeCommandHandler.cs:26]: No problem with this line, but the file has some unused usings

File: src/InnerApi/Controllers/PriceHistoryController.cs [hunk 2]
```diff
@@ -34,13 +34,21 @@ public PriceHistoryController(IQueryDispatcher queryDispatcher, ICommandDispatch
         /// Create apprenticeship price change
         /// </summary>
         /// <param name="apprenticeshipKey">The unique identifier of the apprenticeship</param>
-        /// <param name="request">Details of the requested price change</param>
-        /// <returns>Ok result</returns>
+        /// <param name="request">Details of the requested price change. NB one of either employer or provider id should be included to indicate the initiator of the request</param>
+        /// <returns>Ok on success, Bad Request if both or neither employer & provider id are provided</returns>
```
Comment (MikeV-TalentConsulting) [inline src/InnerApi/Controllers/PriceHistoryController.cs:38]: using an ampersand causes a build warning 'Warning	CS1570	XML comment has badly formed XML'

File: src/InnerApi/Controllers/PriceHistoryController.cs [hunk 3]
```diff
@@ -34,13 +34,21 @@ public PriceHistoryController(IQueryDispatcher queryDispatcher, ICommandDispatch
...
         [HttpPost("{apprenticeshipKey}/priceHistory")]
         [ProducesResponseType(200)]
         public async Task<IActionResult> CreateApprenticeshipPriceChange(Guid apprenticeshipKey, [FromBody] PostCreateApprenticeshipPriceChangeRequest request)
         {
-            await _commandDispatcher.Send(new CreateApprenticeshipPriceChangeRequest(request.ProviderId, request.EmployerId, apprenticeshipKey, request.UserId, request.TrainingPrice, request.AssessmentPrice, request.TotalPrice, request.Reason, request.EffectiveFromDate));
+            try
+            {
+                await _commandDispatcher.Send(new CreateApprenticeshipPriceChangeRequest(request.Requester, apprenticeshipKey, request.UserId, request.TrainingPrice, request.AssessmentPrice, request.TotalPrice, request.Reason, request.EffectiveFromDate));
+            }
+            catch (ArgumentException)
```
Comment (MikeV-TalentConsulting) [inline src/InnerApi/Controllers/PriceHistoryController.cs:47]: Is it worth adding some logging here and in the underlying methods?

File: src/Command/AddPriceHistory/CreateApprenticeshipPriceChangeCommandHandler.cs (hunk 1 above)
Comment (tom-gough) [reply src/Command/AddPriceHistory/CreateApprenticeshipPriceChangeCommandHandler.cs:26]: sorted

File: src/InnerApi/Controllers/PriceHistoryController.cs (hunk 2 above)
Comment (tom-gough) [reply src/InnerApi/Controllers/PriceHistoryController.cs:38]: good spot this is also out of date given the design change we made, will update

File: src/InnerApi/Controllers/PriceHistoryController.cs (hunk 3 above)
Comment (tom-gough) [reply src/InnerApi/Controllers/PriceHistoryController.cs:47]: added

**Review by MikeV-TalentConsulting — APPROVED** [review]

**Pull request merged**